packages/browser/coretest/__tests__/StitchAppClientIntTests.ts

```diff
@@ -520,6 +524,36 @@ describe("StitchAppClient", () => {
     expect(client.auth.listUsers().length).toBe(2);
   });
 
+  it("should authenticate using a custom function", async () => {
+    const { app: appResponse, appResource: app } = await harness.createApp();
+
+      const def = await app.functions.create(new StitchFunction(
+          "funkyAuth",
+          false,
+          `
+          exports = function(payload) {
+              return "foo";
+          }`
+      ))
+
+      await harness.addProvider(app, new FunctionProvider(new FunctionProviderConfig(def.id!!, def.name)));
+
+      const client = harness.getAppClient(appResponse);
+
+      const user = await client.auth.loginWithCredential(new FunctionCredential({"id": "123abc"}));
+
+      expect(user).toBeDefined();
+
+      expect(user.id).toBeDefined();
+      expect(user.identities[0].id).toBeDefined();
+      expect("foo").toEqual(user.identities[0].id);
+      expect(FunctionAuthProvider.DEFAULT_NAME).toEqual(user.loggedInProviderName);
+      expect(FunctionAuthProvider.TYPE).toEqual(user.loggedInProviderType);
+      expect(UserType.Normal).toEqual(user.userType);
+      expect(FunctionAuthProvider.TYPE).toEqual(user.identities[0].providerType);
+      expect(client.auth.isLoggedIn).toBeTruthy();
+  });
+
```

Comment:
can we also add some fail tests as well as tests that actually use whats in the payload?